src/org/loklak/harvester/BaseScraper.java

```diff
@@ -0,0 +1,121 @@
+package org.loklak.harvester;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.lang.StringBuilder;
+import org.loklak.server.AbstractAPIHandler;
+import org.loklak.data.DAO;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.Date;
+import org.loklak.http.ClientConnection;
+import org.json.JSONObject;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.loklak.objects.ProviderType;
+import org.loklak.objects.SourceType;
+import org.loklak.objects.Timeline;
+import org.loklak.objects.QueryEntry;
+import java.util.HashMap;
+import javax.servlet.ServletException;
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.loklak.tools.storage.JSONObjectWithDefault;
+import org.json.JSONObject;
+import org.loklak.data.DAO;
+import org.loklak.http.ClientConnection;
+import org.loklak.server.*;
+import java.io.IOException;
+import org.loklak.http.RemoteAccess;
+import org.loklak.tools.UTF8;
+
+/**
+ * @author vibhcool (Vibhor Verma)
+ * @version 0.1
+ * @since 07.06.2017
+ *
+ * This is Base-Interface for Search-Scrappers.
+ */
+public abstract class BaseScraper extends AbstractAPIHandler {
+
+    // a time stamp that is given by loklak upon the arrival of the post which is the current local time
+    //TODO: check if UTC time needed
+    protected String scraperName;
+    protected String html;
+    protected String url;
+    protected String baseUrl;
+    protected String midUrl = "";
```

Comment:
`MessageEntry` object can be compared to `Post` and `QuoraPost` to `TwitterTweet`. Post is generalized for all scrapers. Every scraper can have it's own specialized object. Post can be used to add all data in same message queue and Timeline iterator

in your example, `baseUrl="https://github.com"`, midUrl="" and query="singhpratyush" , that will be fetched from loklak api endpoint and added to `url` by `githubscraper`